lib/rules/no-extra-parens.js

```diff
@@ -395,13 +413,14 @@ module.exports = {
          */
         function dryBinaryLogical(node) {
             const prec = precedence(node);
-            const shouldSkipLeft = NESTED_BINARY && (node.left.type === "BinaryExpression" || node.left.type === "LogicalExpression");
+            const shouldSkipLeft = (NESTED_BINARY && (node.left.type === "BinaryExpression" || node.left.type === "LogicalExpression")) ||
+              node.left.type === "UnaryExpression" && node.operator === "**";
             const shouldSkipRight = NESTED_BINARY && (node.right.type === "BinaryExpression" || node.right.type === "LogicalExpression");
 
-            if (!shouldSkipLeft && hasExcessParens(node.left) && precedence(node.left) >= prec) {
+            if (!shouldSkipLeft && hasExcessParens(node.left) && (precedence(node.left) > prec || (precedence(node.left) === prec && node.operator !== "**"))) {
```

Comment:
Instead of adding a special case here for `**`, I think it would be better to update the precedence function in `ast-utils` to give `**` a higher precedence. (The function already gives `*` and `/` a higher precedence than `+` and `-`.)